Mathlib/Analysis/NormedSpace/Star/Spectrum.lean

```diff
@@ -122,48 +122,64 @@ theorem selfAdjoint.val_re_map_spectrum [StarModule ℂ A] (a : selfAdjoint A) :
 
 end ComplexScalars
 
-namespace StarAlgHom
-
-variable {F A B : Type*} [NormedRing A] [NormedAlgebra ℂ A] [CompleteSpace A] [StarRing A]
-  [CstarRing A] [NormedRing B] [NormedAlgebra ℂ B] [CompleteSpace B] [StarRing B] [CstarRing B]
-  [FunLike F A B] [AlgHomClass F ℂ A B] [StarAlgHomClass F ℂ A B] (φ : F)
-
-/-- A star algebra homomorphism of complex C⋆-algebras is norm contractive. -/
-theorem nnnorm_apply_le (a : A) : ‖(φ a : B)‖₊ ≤ ‖a‖₊ := by
-  suffices ∀ s : A, IsSelfAdjoint s → ‖φ s‖₊ ≤ ‖s‖₊ by
-    exact nonneg_le_nonneg_of_sq_le_sq zero_le' <| by
-      simpa only [nnnorm_star_mul_self, map_star, map_mul]
-      using this _ (IsSelfAdjoint.star_mul_self a)
-  intro s hs
-  simpa only [hs.spectralRadius_eq_nnnorm, (hs.starHom_apply φ).spectralRadius_eq_nnnorm,
-    coe_le_coe] using
-    show spectralRadius ℂ (φ s) ≤ spectralRadius ℂ s from
-      iSup_le_iSup_of_subset (AlgHom.spectrum_apply_subset φ s)
-#align star_alg_hom.nnnorm_apply_le StarAlgHom.nnnorm_apply_le
-
-/-- A star algebra homomorphism of complex C⋆-algebras is norm contractive. -/
-theorem norm_apply_le (a : A) : ‖(φ a : B)‖ ≤ ‖a‖ :=
-  nnnorm_apply_le φ a
-#align star_alg_hom.norm_apply_le StarAlgHom.norm_apply_le
-
-/-- Star algebra homomorphisms between C⋆-algebras are continuous linear maps.
+namespace NonUnitalStarAlgHom
+
+variable {F A B : Type*}
+variable [NonUnitalNormedRing A] [CompleteSpace A] [StarRing A] [CstarRing A]
+variable [NormedSpace ℂ A] [IsScalarTower ℂ A A] [SMulCommClass ℂ A A] [StarModule ℂ A]
+variable [NonUnitalNormedRing B] [CompleteSpace B] [StarRing B] [CstarRing B]
+variable [NormedSpace ℂ B] [IsScalarTower ℂ B B] [SMulCommClass ℂ B B] [StarModule ℂ B]
+variable [FunLike F A B] [NonUnitalAlgHomClass F ℂ A B] [NonUnitalStarAlgHomClass F ℂ A B]
+
+open Unitization
+
+/-- A non-unital star algebra homomorphism of complex C⋆-algebras is norm contractive. -/
+lemma nnnorm_apply_le (φ : F) (a : A) : ‖φ a‖₊ ≤ ‖a‖₊ := by
+  have h (ψ : Unitization ℂ A →⋆ₐ[ℂ] Unitization ℂ B) (x : Unitization ℂ A) :
+      ‖ψ x‖₊ ≤ ‖x‖₊ := by
+    suffices ∀ s : Unitization ℂ A, IsSelfAdjoint s → ‖ψ s‖₊ ≤ ‖s‖₊ by
+      set_option maxSynthPendingDepth 2 in -- see: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/weird.20type.20class.20synthesis.20error/near/421224482
```

Comment:
In that Zulip thread, the issue is apparently the fact that we can't play with reducibility and so we're stuck. But this was back in February, and now we have `seal` and `unseal` -- do you think we could revisit the issue?

Reply:
I've refactored so that these `synthPending`s are no longer necessary (although it's not my favorite solution, as things are a bit brittle). I don't see how we can use `seal` / `unseal` to help us here, but I've asked Anne on Zulip and pinged you in the message.